### PR TITLE
Curvilinear diffusion validation experiments

### PR DESCRIPTION
**Pull request by glwagner**
This PR implements validation experiments for `HorizontallyCurvilinearAnisotropicDissipation`.

There are three experiments:

1. `validation/curvilinear_diffusion/meridional_diffusion.jl`: Diffusion of a tracer and zonal momentum along a meridian
2. `validation/curvilinear_diffusion/longitudinal_tracer_diffusion.jl`: Diffusion of a tracer along a circle of constant latitude. We would need a non-trivial `NormalFlow` boundary condition to diffuse meridional momentum, so I didn't implement momentum diffusion for this experiment.
3. `validation/curvilinear_diffusion/spot_tracer_diffusion.jl`: Diffusion of a Gaussian spot in longitude and latitude.

# Meridional diffusion 

https://user-images.githubusercontent.com/15271942/109988157-42787180-7cd5-11eb-8dfe-b6f173c6c623.mp4

# Longitudinal tracer diffusion

https://user-images.githubusercontent.com/15271942/109988246-5a4ff580-7cd5-11eb-8f91-f089a3760c7f.mp4

# Spot tracer diffusion

https://user-images.githubusercontent.com/15271942/109988291-650a8a80-7cd5-11eb-81a7-762b65a8d39f.mp4

I propose we merge these and, like the other curvilinear validation experiments, save a quantitative validation for the future. For tracers, quantitative validation requires writing down analytical solutions to the diffusion equation on a spherical surface. For momentum, we cannot yet implement quantitative validation experiments, because our diffusion operator is not correct for the diffusion of a vector on a spherical surface (cc @jm-c, @kburns).

### CODE REVIEW

File: validation/curvilinear_diffusion/longitudinal_tracer_diffusion.jl [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+using Oceananigans
+using Oceananigans.TurbulenceClosures: HorizontallyCurvilinearAnisotropicDiffusivity
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: HydrostaticFreeSurfaceModel
+
+using Statistics
+using JLD2
+using Printf
+using GLMakie
+
+include("../solid_body_rotation/hydrostatic_prescribed_velocity_fields.jl")
```
Comment (ali-ramadhan) [inline validation/curvilinear_diffusion/longitudinal_tracer_diffusion.jl:12]:
```suggestion
include(joinpath(@__DIR__, "..", "solid_body_rotation", "hydrostatic_prescribed_velocity_fields.jl"))
```

File: validation/curvilinear_diffusion/meridional_diffusion.jl [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+lines!(cax, c, ϕ, color = :black)
+using Oceananigans.TurbulenceClosures: HorizontallyCurvilinearAnisotropicDiffusivity
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: HydrostaticFreeSurfaceModel, VectorInvariant
+
+using Statistics
+using JLD2
+using Printf
+using GLMakie
+
+include("../solid_body_rotation/hydrostatic_prescribed_velocity_fields.jl")
```
Comment (ali-ramadhan) [inline validation/curvilinear_diffusion/meridional_diffusion.jl:13]:
```suggestion
include(joinpath(@__DIR__, "..", "solid_body_rotation", "hydrostatic_prescribed_velocity_fields.jl"))
```

File: validation/curvilinear_diffusion/spot_tracer_diffusion.jl [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+using Oceananigans
+using Oceananigans.TurbulenceClosures: HorizontallyCurvilinearAnisotropicDiffusivity
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: HydrostaticFreeSurfaceModel
+
+using Statistics
+using JLD2
+using Printf
+using GLMakie
+
+include("../solid_body_rotation/hydrostatic_prescribed_velocity_fields.jl")
```
Comment (ali-ramadhan) [inline validation/curvilinear_diffusion/spot_tracer_diffusion.jl:12]:
```suggestion
include(joinpath(@__DIR__, "..", "solid_body_rotation", "hydrostatic_prescribed_velocity_fields.jl"))
```

**Review by ali-ramadhan — APPROVED** [review]
I'm okay with merging an eyeball test for now. Just suggested using absolute filepaths which help when running these scripts from CI.

**Review by francispoulin — APPROVED** [review]
The results look reasonable and nothing in the code jumps out as me as being problematic.

**Pull request merged**